lib/TPP/Dialect/Tpp/TppUtils.cpp

```diff
@@ -343,22 +345,41 @@ static bool isUnaryOp(linalg::GenericOp linalgOp) {
   return false;
 }
 
-static bool hasMaxfZeroOp(linalg::LinalgOp linalgOp) {
+static bool matchReluBody(linalg::LinalgOp linalgOp, OperandInfo &info) {
   if (!isa<linalg::GenericOp>(linalgOp))
     return false;
 
   auto genOp = cast<linalg::GenericOp>(linalgOp);
   if (!hasOnlyOp<arith::MaxFOp>(genOp.getRegion()))
     return false;
 
-  for (Operation &op : genOp.getRegion().front()) {
-    if (auto maxfOp = dyn_cast_or_null<arith::MaxFOp>(op)) {
-      // Check both operands if either one is a zero filled tensor.
-      if (isZeroTensor(maxfOp.getLhs()) || isZeroTensor(maxfOp.getRhs()))
-        return true;
-    }
-  }
+  Operation *regionOp = &genOp.getRegion().front().front();
+  auto maxfOp = cast<arith::MaxFOp>(regionOp);
+  Value maxfLhs = maxfOp.getLhs();
+  Value maxfRhs = maxfOp.getRhs();
+
+  if (isZeroTensor(maxfLhs) && isZeroTensor(maxfRhs))
+    return false;
+
+  if (isZeroTensor(maxfLhs)) {
```

Comment:
This could have been a lambda passed with alternating arguments

Reply:
+1 on unifying/rewriting these two if cases into one